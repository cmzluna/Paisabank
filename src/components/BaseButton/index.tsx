import React from "react";
import { TouchableOpacity } from "react-native";

interface BaseButtonProps {
  fontSize?: number;
  onPress?: () => void;
  children?: JSX.Element;
}

const BaseButton = ({
  fontSize,
  onPress,
  children,
  ...props
}: BaseButtonProps): JSX.Element | null => {
  if (!children) return null;

  return (
    <TouchableOpacity activeOpacity={0.45} onPress={onPress} {...props}>
      {children}
    </TouchableOpacity>
  );
};

export default BaseButton;
