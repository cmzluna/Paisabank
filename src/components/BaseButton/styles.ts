import styled from 'styled-components/native';

interface TextCompProps {
  fontSize?: number;
}

const TextComp = styled.Text`
  color: #fff;
  font-size: ${({fontSize}: TextCompProps) => fontSize ?? 12}px;
  font-family: 'AvenirLTStd-Black';
`;

export {TextComp};
