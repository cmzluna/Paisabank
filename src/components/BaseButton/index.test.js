import {TouchableOpacity} from 'react-native';
import React from 'react';
import renderer from 'react-test-renderer';
import BaseButton from './index';

describe('BaseButton component', () => {
  describe('returns error', () => {
    it('returns null when title is not passed', () => {
      const BaseButtonComponent = renderer
        .create(<BaseButton onPress={() => {}} />)
        .toJSON();
      expect(BaseButtonComponent).toBeNull();
    });
  });

  describe('renders correctly', () => {
    it('match snapshot', () => {
      const BaseButtonComponent = renderer
        .create(<BaseButton title="Ok" onPress={() => {}} />)
        .toJSON();
      expect(BaseButtonComponent).toMatchSnapshot();
    });

    it('renders children type text', () => {
      const BaseButtonComponent = renderer
        .create(<BaseButton title="Text Button" onPress={() => {}} />)
        .toJSON();
      expect(BaseButtonComponent.children[0].type).toBe('Text');
    });

    it('renders children text with value passed as title prop', () => {
      const BaseButtonComponent = renderer
        .create(<BaseButton title="Confirm" onPress={() => {}} />)
        .toJSON();
      expect(BaseButtonComponent.children[0].children[0]).toBe('Confirm');
    });

    it('function passed as prop is called when button is pressed', async () => {
      const mockedFn = jest.fn();
      const BaseButtonComponent = renderer.create(
        <BaseButton title="Ok" onPress={mockedFn} />,
      ).root;

      const TouchableOpacityRendered =
        BaseButtonComponent.findByType(TouchableOpacity);
      TouchableOpacityRendered.props.onPress();
      expect(mockedFn).toHaveBeenCalledTimes(1);
    });
  });
});
