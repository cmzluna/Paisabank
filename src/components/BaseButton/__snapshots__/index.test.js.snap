// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BaseButton component renders correctly match snapshot 1`] = `
<View
  accessibilityState={
    {
      "busy": undefined,
      "checked": undefined,
      "disabled": undefined,
      "expanded": undefined,
      "selected": undefined,
    }
  }
  accessibilityValue={
    {
      "max": undefined,
      "min": undefined,
      "now": undefined,
      "text": undefined,
    }
  }
  accessible={true}
  collapsable={false}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    {
      "opacity": 1,
    }
  }
>
  <Text>
    Button Text
  </Text>
</View>
`;
